Update shadow jar plugin, and code cleanup

Omico opened the pull request:
Please review each commit message for the details.

Omico committed:
Update build.gradle.kts to use new layout API for local repository URL and correct license url

Omico committed:
Remove unused code in gradleUtils.kt

Test.systemProperties() marked as EXTENSION_SHADOWED_BY_MEMBER

Omico committed:
Rename shadow to shadowJar, because the shadow extension

Omico committed:
Update shadow jar plugin

See also, https://github.com/GradleUp/shadow/issues/908

Omico committed:
Refactor deprecated experimental configuration messages

MatkovIvan commented on gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposePlugin.kt:
```diff
@@ -65,44 +63,44 @@ abstract class ComposePlugin : Plugin<Project> {
+        val runtime = composeDependency("org.jetbrains.compose.runtime:runtime")
+        val runtimeSaveable = composeDependency("org.jetbrains.compose.runtime:runtime-saveable")
+        val ui = composeDependency("org.jetbrains.compose.ui:ui")
+        val uiBackHandler = composeDependency("org.jetbrains.compose.ui:ui-backhandler")
```
```suggestion
        @ExperimentalComposeLibrary
        val uiBackHandler = composeDependency("org.jetbrains.compose.ui:ui-backhandler")
```

Bigray2025 reviewed (commented):
I can I get the app coded here in my iOS device??

Omico committed:
Remove `get()` and lazy for all dependencies, and sort all dependencies

Since ComposeBuildConfig.composeVersion is a static Sting and its parent is an object class we don't need to use get or lazy.

MatkovIvan commented on gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposePlugin.kt:
```diff
@@ -65,28 +63,34 @@ abstract class ComposePlugin : Plugin<Project> {
+        val uiTestJUnit4 = composeDependency("org.jetbrains.compose.ui:ui-test-junit4")
+
+        @ExperimentalComposeLibrary
+        val uiBackHandler = composeDependency("org.jetbrains.compose.ui:ui-backhandler")
```
I'd prefer not adding such new API at all.
1. We're not super happy of having such aliases at all. Probably most relevant task about it is [CMP-5861](https://youtrack.jetbrains.com/issue/CMP-5861)
2. We don't have plans to stabilize this module at all. It's opposite - it will be replaced to [navigationevent](https://developer.android.com/jetpack/androidx/releases/navigationevent) soon-ish

MatkovIvan reviewed (commented):
Combining cleanups and introducing new public API is not good idea.

PS I've assigned right people as reviewers

terrakok commented:
Could you explain your motivation for refactorings and version updates?
I don't see a reason for that at all. Yes, might be it is not perfect but there is no reasons to change it right now.
